raft/node_util_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2015 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raft
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+)
+
+type nodeTestHarness struct {
+	*node
+	t *testing.T
```

Comment:
```suggestion
	*testing.T
```

Reply:
After this change, you can remove all the following methods (e.g. Debug(f), Error(f), Info(f), Warning(f), Fatal(f), Panic(f)

Reply:
No, we are implementing `raft.Logger` here, `*testing.T` does not implement that. It is true that some methods overlap (for example, `*testing.T` has a matching `Fatalf`) but semantically we don't necessarily want to call `t.Fatalf` just because a raft node calls `Fatalf`. I will leave this as is.